### DISCUSSION TITLE
Vuex in Inerja

### DISCUSSION THREAD

**Discussion by CarlsonBuma** [inertiajs/inertia#869] [Help]
Hello,

I try to implement a global store by Vuex. 
How to install / implement it?

As I cant find any documentation, I would like to ask for your help.

Thanks alot!

**Comment by reinink** [comment]
As a general rule, we don't recommend using Vuex with Inertia. One of the key benefits of using Inertia is that you avoid client-side state entirely, and keep all state server-side. 👍

**Reply by CarlsonBuma** [reply]
Thank you. its Used for Frontend-Design purpose. I found out how to: in app.js

import {createStore} from 'vuex';
const store = createStore({
    //
})

reateInertiaApp({
    title: (title) => `${title} - ${appName}`,
    resolve: (name) => require(`./Pages/${name}.vue`),
    setup({ el, app, props, plugin }) {
        return createApp({ render: () => h(app, props) })
            .use(plugin)
            .use(store)
            .mixin({ methods: { route } })
            .mount(el);
    },
});
